runtime/include/wasm_memory.h

```diff
@@ -42,20 +44,21 @@ wasm_memory_allocate(size_t initial, size_t max)
 
 	int rc = mprotect(self, size_to_read_write, PROT_READ | PROT_WRITE);
 	if (rc != 0) {
-		perror("wasm_memory_allocate - prot r/w failed");
+		perror("wasm_memory_new - prot r/w failed");
 		munmap(self, size_to_alloc);
 		assert(0);
 		return NULL;
 	}
 
+	ps_list_init_d(self);
 	self->size     = initial;
 	self->capacity = initial;
 	self->max      = max;
 	return self;
 }
 
 static inline void
-wasm_memory_free(struct wasm_memory *self)
+wasm_memory_delete(struct wasm_memory *self)
```

Comment:
The problem with using C++ naming for a lot of this stuff is that it could lull people into thinking that new and delete function as they do in C++. They do not as both of those operations are quite complex in C++. I'm *guessing* this is why it is rarer to use this naming in C. Conflating ideas can lead to bugs.

Reply:
What else would call "allocate the outermost data structure" and "run the initializer/constructor which allocates members with RAII semantics" (and visa-versa with the deinit/dtor and free)? A lot of the existing APIs were simpler because the existing code made assumptions about being a struct nested as a member in another struct or the buffer being static or other such things?

I think there's a big win to having some opinions about "OOP-like" translation units in our project and the lifecycle of how those work. This is similar to the approach in this book: http://www.r-5.org/files/books/computers/languages/c/mod/David_R_Hanson-C_Interfaces_and_Implementations-EN.pdf